var/spack/repos/builtin/packages/gptune/package.py

```diff
@@ -0,0 +1,173 @@
+                envfile.write('software_json=$(echo ",\\\"software_configuration\\\":' +
+                              '{\\\"' + spec['blas'].name +
+                              '\\\":{\\\"version_split\\\":' +
+                              ' [' + str(spec['blas'].versions).replace('.', ',') +
+                              ']},\\\"' + spec['mpi'].name +
+                              '\\\":{\\\"version_split\\\": [' +
+                              str(spec['mpi'].versions).replace('.', ',') + ']},\\\"' +
+                              spec['scalapack'].name +
+                              '\\\":{\\\"version_split\\\": [' +
+                              str(spec['scalapack'].versions).replace('.', ',') +
+                              ']},\\\"' +
+                              str(comp_name) + '\\\":{\\\"version_split\\\": [' +
+                              str(comp_version) + ']}}") \n')
+                envfile.write('loadable_software_json=$(echo ",\\\"loadable_software_' +
+                              'configurations\\\":{\\\"' + spec['blas'].name +
+                              '\\\":{\\\"version_split\\\": [' +
+                              str(spec['blas'].versions).replace('.', ',') +
+                              ']},\\\"' + spec['mpi'].name +
+                              '\\\":{\\\"version_split\\\": [' +
+                              str(spec['mpi'].versions).replace('.', ',') + ']},\\\"' +
+                              spec['scalapack'].name +
+                              '\\\":{\\\"version_split\\\": [' +
+                              str(spec['scalapack'].versions).replace('.', ',') +
+                              ']},\\\"' + str(comp_name) +
+                              '\\\":{\\\"version_split\\\": ['
+                              + str(comp_version) + ']}}") \n')
+                envfile.write('machine_json=$(echo ",\\\"machine_configuration\\\":' +
+                              '{\\\"machine_name\\\":\\\"$machine\\\",\\\"$proc\\\":' +
+                              '{\\\"nodes\\\":$nodes,\\\"cores\\\":$cores}}") \n')
+                envfile.write('loadable_machine_json=$(echo ",\\\"loadable_machine_' +
+                              'configurations\\\":{\\\"$machine\\\":{\\\"$proc\\\":' +
+                              '{\\\"nodes\\\":$nodes,\\\"cores\\\":$cores}}}") \n')
+
+        if '+app' in spec:
+            apps = ['GPTune-Demo', 'SuperLU_DIST', 'SuperLU_DIST_RCI',
+                    'Scalapack-PDGEQRF', 'Scalapack-PDGEQRF_RCI']
+        else:
+            apps = ['GPTune-Demo', 'Scalapack-PDGEQRF', 'Scalapack-PDGEQRF_RCI']
+
+        for app in apps:
+            with working_dir(join_path(test_dir, app), create=False):
+                # PDGEQRF with GPTune
+                self.run_test('bash', options=['run_examples.sh'], work_dir='.',
+                              purpose='gptune smoke test for {0}'.format(app))
```

Comment:
I really like the tightness of this code but you pass the directory in `working_dir()` as the `work_dir` (assuming that is appropriate for this case).

Reply:
Thanks I will change this in the next PR